src/include/duckdb/optimizer/join_order_optimizer.hpp

```diff
@@ -72,18 +58,29 @@ class JoinOrderOptimizer {
 	//! C}
 	expression_map_t<vector<FilterInfo *>> equivalence_sets;
 
+	CardinalityEstimator cardinality_estimator;
+
+	bool full_plan_found;
+	bool must_update_full_plan;
+	unordered_set<string> join_nodes_in_full_plan;
```

Comment:
Should this be a string? Can't we create an `unordered_set` on the relation node by defining a hash/equality function?